Improve progress report on zeros_mt and generateRandom

Algunenano opened the pull request:
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Improve progress report on zeros_mt and generateRandom

### Documentation entry for user-facing changes

Closes https://github.com/ClickHouse/ClickHouse/issues/63834

Algunenano committed:
Simplify 03023_zeros_generate_random_with_limit_progress_bar

robot-ch-test-poll4 commented:
<!-- automatic status comment for PR #64804 from Algunenano/ClickHouse:simplify_03023_zeros_generate_random_with_limit_progress_bar -->
*This is an automated comment for commit 3d7beae8fd4140f90917a592f3aae1bfecb90c5a with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64804/3d7beae8fd4140f90917a592f3aae1bfecb90c5a/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

alexey-milovidov commented:
It's still flaky.

Algunenano committed:
Better debugging

Algunenano committed:
Stabilize and adjust number of streams

Algunenano commented:
Changed a few extra things (moving to an improvement):
* Adjusted number of streams based on limit
* Made total_rows_to_read be reported by the first thread reading data, not the first stream created (which might or might not read any data depending on how fast it is)

Algunenano committed:
Fix number of stream decision

Algunenano committed:
Compiler complains

Algunenano committed:
Compiler happy. Developer unhappy
